src/Microsoft.NetCore.Analyzers/Core/Runtime/DisposableFieldsShouldBeDisposed.cs

```diff
@@ -233,13 +241,13 @@ bool ShouldAnalyze(INamedTypeSymbol namedType)
                     // We only want to analyze types which are disposable (implement System.IDisposable directly or indirectly)
                     // and have at least one disposable field.
                     return !hasErrors &&
-                        namedType.IsDisposable(disposeAnalysisHelper.IDisposable) &&
+                        namedType.IsDisposable(disposeAnalysisHelper!.IDisposable) &&
                         !disposeAnalysisHelper.GetDisposableFields(namedType).IsEmpty &&
                         !namedType.IsConfiguredToSkipAnalysis(compilationContext.Options, Rule, compilationContext.Compilation, compilationContext.CancellationToken);
                 }
 
                 bool IsDisposeMethod(IMethodSymbol method)
-                    => method.GetDisposeMethodKind(disposeAnalysisHelper.IDisposable, disposeAnalysisHelper.Task) != DisposeMethodKind.None;
+                    => method.GetDisposeMethodKind(disposeAnalysisHelper!.IDisposable, disposeAnalysisHelper.Task) != DisposeMethodKind.None;
```

Comment:
💡 Consider making the containing local function static

Reply:
That needs us to end up passing the disposeAnalysisHelper around in all the local functions, which seems redundant and verbose just to satisfy nullablility warnings. I think it would instead be better for compiler's nullability analysis to be enhanced so that this local function can never be called on the code path where `disposeAnalysisHelper` can be null. I'll stick with the suppression for now.